tests/cupy_tests/core_tests/test_internal.py

```diff
@@ -126,10 +126,10 @@ def test_infer(self):
 
     {'slice': (None, 8, -1),  'expect': (9, 8, -1)},
     {'slice': (-3, 6, -1),    'expect': (7, 6, -1)},
-    {'slice': (11, 8, -1),    'expect': (10, 8, -1)},
-    {'slice': (11, 11, -1),   'expect': (10, 10, -1)},
+    {'slice': (11, 8, -1),    'expect': (9, 8, -1)},
+    {'slice': (11, 11, -1),   'expect': (9, 9, -1)},
     {'slice': (-11, 8, -1),   'expect': (0, 0, -1)},
-    {'slice': (-11, -11, -1), 'expect': (0, 0, -1)},
+    {'slice': (-11, -11, -1), 'expect': (0, -1, -1)},
```

Comment:
As indexing is difficult to check, more thorough tests will assure us.
For example, how about adding the test cases where
`slice = (10, 8, -1), (9, 8, -1)` (similar to `(11, 8, -1)`)
`slice = (10, 10, -1), (9, 9, -1)` (similar to `(11, 11, -1)`) 
`slice = (9, 10, -1)`
`slice = (-10, -10, -1), (-9, -9, -1)` (similar to `(-11, -11, -1)`)
`slice = (-10, -9, -1), (-9, -10, -1)`
I think the boundary is between 9 and 10, rather than 10 and 11. Of course we do not have to remove the existing test cases.